presto-main/src/main/java/io/prestosql/server/protocol/QueryResultRows.java

```diff
@@ -218,9 +218,9 @@ private Object getLegacyValue(Object value, Type type)
                 return value;
             }
 
-            return unmodifiableList(((List<Object>) value).stream()
+            return ((List<Object>) value).stream()
                     .map(element -> getLegacyValue(element, elementType))
-                    .collect(toList()));
+                    .collect(toUnmodifiableList());
```

Comment:
Not equivalent. Will not allow `null` elements.

Reply:
Good catch. Reverted.